File: Cuarto/Laboratorio/Talleres/RobocodeLabo/src/Robotito3/CombatStrategy.java
package Robotito3;

import robocode.*;

public abstract class CombatStrategy {
    JuniorRobot robot;
    double bulletHalfPower = 1.5;
    double bulletMaxPower = 3;

    void init(JuniorRobot robot) {
        this.robot = robot;
        onInit();
    }

    abstract void onInit();

    abstract void onTick();

    abstract void onScannedRobot();

    abstract void onHitWall();

    abstract void onHitRobot();

    abstract void onHitByBullet();

}

File: Cuarto/Laboratorio/Talleres/RobocodeLabo/src/robotitoKotlin2/CombatStrategy.kt
package robotitoKotlin2

import robocode.JuniorRobot

abstract class CombatStrategy(protected val robot: JuniorRobot)  {

    var bulletHalfPower: Double = 1.5
    var bulletMaxPower: Double = 3.0

    abstract fun onTick()

    abstract fun onScannedRobot()

    abstract fun onHitWall()

    abstract fun onHitRobot()

    abstract fun onHitByBullet()

}